countries.js

```diff
@@ -0,0 +1,87 @@
+document.addEventListener("DOMContentLoaded", function() {
+    // Fetch country data asynchronously
+    fetchCountryData()
+        .then(countries => {
+            // Update the UI with fetched data
+            displayCountries(countries);
+        })
+        .catch(error => {
+            console.error('Error loading countries:', error);
+        });
+
+    // Add event listener for filter by region dropdown
+    $('#continent-filter').on('change', function() {
+        const selectedContinent = $(this).val();
+        let apiUrl = "https://restcountries.com/v3.1/all";
```

Comment:
Why mix jquery and vanilla js ?

Also, do you know what `this` means in js? why did you used it in here?